File: admin/src/main/java/com/jigumulmi/member/dto/AdminMemberListResponseDto.java
package com.jigumulmi.member.dto;

import com.jigumulmi.common.PagedResponseDto;
import com.jigumulmi.member.domain.Member;
import com.jigumulmi.member.dto.AdminMemberListResponseDto.MemberDto;
import com.jigumulmi.member.dto.response.MemberDetailResponseDto;
import java.time.LocalDateTime;
import lombok.Getter;
import lombok.experimental.SuperBuilder;

@Getter
@SuperBuilder
public class AdminMemberListResponseDto extends PagedResponseDto<MemberDto> {

    @Getter
    @SuperBuilder
    public static class MemberDto extends MemberDetailResponseDto {

        private LocalDateTime modifiedAt;
        private Long KakaoUserId;

        public static MemberDto from(Member member) {
            return MemberDto.builder()
                .createdAt(member.getCreatedAt())
                .modifiedAt(member.getModifiedAt())
                .id(member.getId())
                .nickname(member.getNickname())
                .email(member.getEmail())
                .KakaoUserId(member.getKakaoUserId())
                .deregisteredAt(member.getDeregisteredAt())
                .isAdmin(member.getIsAdmin())
                .build();
        }

    }
}

File: admin/src/main/kotlin/com/jigumulmi/member/dto/AdminMemberListResponseDto.kt
package com.jigumulmi.member.dto

import com.jigumulmi.common.AdminPagedResponseDto
import com.jigumulmi.member.domain.Member
import java.time.LocalDateTime


data class AdminMemberListResponseDto(
    override val page: PageDto,
    override val data: List<MemberDto>
) : AdminPagedResponseDto<AdminMemberListResponseDto.MemberDto>(page, data) {

    data class MemberDto(
        var createdAt: LocalDateTime,
        val modifiedAt: LocalDateTime,
        val deregisteredAt: LocalDateTime? = null,
        val email: String,
        val isAdmin: Boolean,
        val kakaoUserId: Long,
    ) {
        companion object {
            fun from(member: Member): MemberDto {
                return MemberDto(
                    createdAt = member.createdAt,
                    modifiedAt = member.modifiedAt,
                    deregisteredAt = member.deregisteredAt,
                    email = member.email,
                    isAdmin = member.isAdmin,
                    kakaoUserId = member.kakaoUserId
                )
            }
        }
    }
}
